DSpace/dspace-angular
# Search query is not set by default when opening an external sources tab

## References
* Related #2083
* Maintenance PR: #2396

## Description
In a previous PR I made it possible to pass the already filled input value of a relation field to the search bar from the popup modal. That fix only worked for the internal entities tab, this PR now also fixes this for the tab for external sources tabs.

## Instructions for Reviewers
List of changes in this PR:
* Added a new `@Input query: string` to the `DsDynamicLookupRelationExternalSourceTabComponent`
* Passed the `query` that was already filled in the form to the component in `DsDynamicLookupRelationModalComponent`

**Guidance for how to test/review this PR:**
- Add the following config to your submission form config in submission-forms.xml:
   ```xml
   <row>
       <relation-field>
           <relationship-type>isAuthorOfPublication</relationship-type>
           <search-configuration>person</search-configuration>
           <repeatable>true</repeatable>
           <name-variants>true</name-variants>
           <label>Authors</label>
           <hint>Enter the author's name (Family name, Given names).</hint>
           <linked-metadata-field>
               <dc-schema>dc</dc-schema>
               <dc-element>contributor</dc-element>
               <dc-qualifier>author</dc-qualifier>
               <input-type>onebox</input-type>
           </linked-metadata-field>
           <externalsources>orcid</externalsources>
           <required></required>
       </relation-field>
   </row>
   ```
- Type "SMITH" in the author input field & then press the lookup (spyglass) button
- When you select another import source (ORCID in this case) "SMITH" should now also be filled in

## Checklist
